artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/ActiveMQSessionContext.java

```diff
@@ -686,7 +686,7 @@ protected CreateSessionMessage newCreateSession(String username,
                                                    boolean autoCommitSends,
                                                    boolean autoCommitAcks,
                                                    boolean preAcknowledge) {
-      return new CreateSessionMessage(name, sessionChannel.getID(), VersionLoader.getVersion().getIncrementingVersion(), username, password, minLargeMessageSize, xa, autoCommitSends, autoCommitAcks, preAcknowledge, confirmationWindow, null);
+      return new CreateSessionMessage(name, sessionChannel.getID(), serverVersion, username, password, minLargeMessageSize, xa, autoCommitSends, autoCommitAcks, preAcknowledge, confirmationWindow, null);
```

Comment:
The ```newCreateSession()``` is only called in ```recreateSession()``` method, so my reasoning was that the message will be sent to the same server that the client was previously connected to (is that correct?), so the serverVersion that was previously established can be reused.

The alternative could be to start a cycle of CreateSessionMessages with the highest know version and decrease the version if ```ActiveMQIncompatibleClientServerException``` is returned? I think this is how establishing new connection works.